packages/manager/modules/vps/src/upscale/model-price/model-price.controller.js

```diff
@@ -0,0 +1,13 @@
+import { Price } from '@ovh-ux/manager-models';
+import UpscaleService from '../upscale.service';
+
+export default class {
+  $onInit() {
+    UpscaleService.validateLanguage(this.language);
+  }
+
+  buildPrice() {
+    const price = new Price(this.price);
+    return UpscaleService.buildPriceToDisplay(price, this.language);
+  }
```

Comment:
Looks like both controllers http://github.com/ovh/manager/pull/2525/files#diff-e51ca0dx0a33d8b5ffb38ab635b7b0e57R11 are doing the same thing , like the template (except for translation). Maybe we could have a single component in which the translation is injected instead so we don't need to duplicate ?

Reply:
I already answered this question, both components are actually pretty similar but differents in their intents and it would be bad idea to DRY them (more information here: https://www.amazon.com/Pragmatic-Programmer-journey-mastery-Anniversary/dp/0135957052/ref=sr_1_1?crid=3O2JZTUCLVHAS&keywords=pragmatic+programmer&qid=1584952852&sprefix=pragma%2Caps%2C243&sr=8-1 😄 )

Reply:
I'm missing context here, so what I see is only two component doing the exact same thing. Let's say I want to reuse one of them, I won't be able to determine the one suiting my needs. 
If they are meant to evolve differently but still have a common base maybe we can temporarily have an inheritance from a parent controller or something that will avoid duplicating this logic

Reply:
It would not be the right abstraction, they look like the same but they are not :)